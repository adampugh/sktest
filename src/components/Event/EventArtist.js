import React from 'react';
import { Link } from 'react-router-dom';

const EventArtist = ({ name, img, id }) => (
    <div className="eventArtist">
        <img className="eventArtist__img" src={img} alt={name} />
        <div className="eventArtist__text">
            <h1 className="title">{name}</h1>
            <Link to={`/artist/${id}`}>
                <button className="btn">view artist</button>
            </Link>
        </div>
    </div>
);

export default EventArtist;