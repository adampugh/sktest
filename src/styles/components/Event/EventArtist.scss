.eventArtist {
    display: flex;
    align-content: center;
    padding: 20px 0;
    border-bottom: 2px solid lightgrey;
    img {
        height: 100px;
        margin-right: 10px;
    }
}

.eventArtist__text {
    align-self: center;
}