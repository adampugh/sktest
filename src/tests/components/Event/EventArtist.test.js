import React from 'react';
import '../../setupTests';
import { shallow } from 'enzyme';

import EventArtist from '../../../components/Event/EventArtist';

describe("<EventArtist />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<EventArtist name={'bob'} img={'img'} id={'123'} />);
        expect(wrapper).toMatchSnapshot();
    });
});