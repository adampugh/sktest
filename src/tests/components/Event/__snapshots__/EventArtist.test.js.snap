// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Artist /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventArtist
    id="123"
    img="img"
    name="bob"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="bob"
          className="eventArtist__img"
          src="img"
        />,
        <div
          className="eventArtist__text"
        >
          <h1
            className="title"
          >
            bob
          </h1>
          <Link
            to="/artist/123"
          >
            <button
              className="btn"
            >
              view artist
            </button>
          </Link>
        </div>,
      ],
      "className": "eventArtist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "bob",
          "className": "eventArtist__img",
          "src": "img",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
            >
              bob
            </h1>,
            <Link
              to="/artist/123"
            >
              <button
                className="btn"
              >
                view artist
              </button>
            </Link>,
          ],
          "className": "eventArtist__text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "bob",
              "className": "title",
            },
            "ref": null,
            "rendered": "bob",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn"
              >
                view artist
              </button>,
              "to": "/artist/123",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "view artist",
                "className": "btn",
              },
              "ref": null,
              "rendered": "view artist",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="bob"
            className="eventArtist__img"
            src="img"
          />,
          <div
            className="eventArtist__text"
          >
            <h1
              className="title"
            >
              bob
            </h1>
            <Link
              to="/artist/123"
            >
              <button
                className="btn"
              >
                view artist
              </button>
            </Link>
          </div>,
        ],
        "className": "eventArtist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "bob",
            "className": "eventArtist__img",
            "src": "img",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="title"
              >
                bob
              </h1>,
              <Link
                to="/artist/123"
              >
                <button
                  className="btn"
                >
                  view artist
                </button>
              </Link>,
            ],
            "className": "eventArtist__text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "bob",
                "className": "title",
              },
              "ref": null,
              "rendered": "bob",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn"
                >
                  view artist
                </button>,
                "to": "/artist/123",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "view artist",
                  "className": "btn",
                },
                "ref": null,
                "rendered": "view artist",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<EventArtist /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventArtist
    id="123"
    img="img"
    name="bob"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="bob"
          className="eventArtist__img"
          src="img"
        />,
        <div
          className="eventArtist__text"
        >
          <h1
            className="title"
          >
            bob
          </h1>
          <Link
            to="/artist/123"
          >
            <button
              className="btn"
            >
              view artist
            </button>
          </Link>
        </div>,
      ],
      "className": "eventArtist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "bob",
          "className": "eventArtist__img",
          "src": "img",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
            >
              bob
            </h1>,
            <Link
              to="/artist/123"
            >
              <button
                className="btn"
              >
                view artist
              </button>
            </Link>,
          ],
          "className": "eventArtist__text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "bob",
              "className": "title",
            },
            "ref": null,
            "rendered": "bob",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn"
              >
                view artist
              </button>,
              "to": "/artist/123",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "view artist",
                "className": "btn",
              },
              "ref": null,
              "rendered": "view artist",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="bob"
            className="eventArtist__img"
            src="img"
          />,
          <div
            className="eventArtist__text"
          >
            <h1
              className="title"
            >
              bob
            </h1>
            <Link
              to="/artist/123"
            >
              <button
                className="btn"
              >
                view artist
              </button>
            </Link>
          </div>,
        ],
        "className": "eventArtist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "bob",
            "className": "eventArtist__img",
            "src": "img",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="title"
              >
                bob
              </h1>,
              <Link
                to="/artist/123"
              >
                <button
                  className="btn"
                >
                  view artist
                </button>
              </Link>,
            ],
            "className": "eventArtist__text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "bob",
                "className": "title",
              },
              "ref": null,
              "rendered": "bob",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn"
                >
                  view artist
                </button>,
                "to": "/artist/123",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "view artist",
                  "className": "btn",
                },
                "ref": null,
                "rendered": "view artist",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
